frame/support/src/lib.rs

```diff
@@ -84,8 +84,24 @@ pub use sp_runtime::{self, ConsensusEngineId, print, traits::Printable};
 #[derive(Debug)]
 pub enum Never {}
 
-/// Macro for easily creating a new implementation of the `Get` trait. If `const` token is used, the
-/// rhs of the expression must be `const`-only, and get is implemented as `const`:
+/// Create new implementations of the [`Get`](crate::traits::Get) trait.
+///
+/// The so-called parameter type can be created in three different ways:
+///
+/// - Using `const` to create a parameter type that provides a `const` getter.
+///   It is required that the `value` is const.
+///
+/// - Declare the parameter type without `const` to have more freedom when creating the value.
+///
+/// - Using `storage` to create a storage parameter type. This type is special as it tries to
+///   load the value from the storage under a fixed key. If the value could not be found in the
+///   storage, the given default value will be returned. It is required that the value implements
+///   [`Encode`](codec::Encode) and [`Decode`](codec::Decode). The key for looking up the value
+///   in the storage is built using the following formular:
+///
+///   `twox_128(":" ++ NAME ++ ":")` where `NAME` is the name that is passed as type name.
```

Comment:
we have each module at `twox_128(modulename)` indeed having `:` prefixed and suffixed allow their name not to collide. looks fine but I wonder what is best.